presto-main/src/main/java/io/prestosql/testing/TestingAccessControlManager.java

```diff
@@ -514,6 +516,18 @@ public void checkCanExecuteFunction(SecurityContext context, String functionName
         return super.getColumnMasks(context, tableName, column, type);
     }
 
+    @Override
+    public Set<SchemaTableName> filterTables(SecurityContext context, String catalogName, Set<SchemaTableName> tableNames)
+    {
+        if (shouldDenyPrivilege(context.getIdentity().getUser(), catalogName, FILTER_TABLES)) {
```

Comment:
I don't think you should use `denyPrivileges` here. This case is very different. Notice how `filterCatalogs` is implemented and follow same pattern.